xds/src/test/java/io/grpc/xds/SharedXdsClientPoolProviderTest.java

```diff
@@ -138,4 +157,71 @@ public void refCountedXdsClientObjectPool_getObjectCreatesNewInstanceIfAlreadySh
     assertThat(xdsClient2).isNotSameInstanceAs(xdsClient1);
     xdsClientPool.returnObject(xdsClient2);
   }
+
+  private class CallCredsServerInterceptor implements ServerInterceptor {
+    private String token;
+    private SettableFuture<Void> requestDone = SettableFuture.create();
+
+    @Override
+    public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
+        ServerCall<ReqT, RespT> serverCall,
+        Metadata metadata,
+        ServerCallHandler<ReqT, RespT> next) {
+      String callCreds = metadata.get(AUTHORIZATION_METADATA_KEY);
+      if (callCreds != null) {
+        token = callCreds.substring("Bearer".length()).trim();
```

Comment:
Instead of parsing this, just keep `token` as-is and in the assertion do `isEqualTo("Bearer token")`. More obvious and less bug prone. Also, it handles errors more clearly (e.g., if it wasn't prefixed with Bearer) it'd fail the test thread instead of hanging (because this substring() would throw).